runners/samza/src/main/java/org/apache/beam/runners/samza/runtime/SamzaTimerInternalsFactory.java

```diff
@@ -307,7 +327,11 @@ public void setTimer(TimerData timerData) {
           break;
 
         case PROCESSING_TIME:
-          timerRegistry.schedule(keyedTimerData, timerData.getTimestamp().getMillis());
+          // Append to buffer iff not full
+          if (processTimeBuffer.size() < maxProcessTimerBufferSize) {
+            processTimeBuffer.add(keyedTimerData);
+          }
```

Comment:
Do we need to do anything in case we have reached the buffer size? just like the event time scenario.

Reply:
No we don't need to replace any events in the buffer like in event time because process time is fifo only, if the buffer is full we don't do anything and it is simply stored in the state